fix(devtools): add event tagging to prevent DDOS

JeanMeche opened the pull request:
One common problem encountered by the devtools content script is that it accepted almost any message send over the message bus. Some websites like `auth.openai.com` were spamming the bus and DDOS the devtools app.

By introducing event tagging and skipping non-devtools events we prevent DDOS of the Angular devtools content script by on forward tagged events. 

fixes #62471 #62450 #55854

JeanMeche commented:
/gemini

JeanMeche commented:
/gemini review

AleksanderBodurri reviewed (approved):
So happy to see this issue solved. Thanks for this @JeanMeche 🙏

JeanMeche committed:
fix(devtools): add event tagging to prevent DDOS

One common problem encountered by the devtools content script is that it accepted almost any message send over the message bus. Some websites like `auth.openai.com` were spamming the bus and DDOS the devtools app.

By introducing event tagging and skipping non-devtools events we prevent DDOS of the Angular devtools content script by on forward tagged events.

JeanMeche commented on devtools/projects/shell-browser/src/app/content-script.ts:
```diff
@@ -94,20 +95,58 @@ if (!backendInitialized) {
+
+// This is where we listen for messages sent from the Angular DevTools extension to the content script.
+// We have some logic to handle spam protection and event tagging.
+fromEvent<MessageEvent>(window, 'message')
```
When I initially wrote the fix I first when for the bailout. But I wasn't instantly effective (you still felt that freeze until the bailout happened) so I introduced the event tagging. 

And you're right, event tagging is enough for preventing external messages from DDOS the extension/browsers. So I dropped the bailout. This allows us to keep a much simpler code.
